### PR TITLE
[GC@Init] Sorcery Gem

### PR DESCRIPTION
**Pull request by pinifloyd**
Basic implementation of authentication service using sorcery gem. It's basic implementation, that mean default settings, so we should extend / change it later when it will needed (ex: remember_me...).

/cc @ErrorSoul @rkichan @r0ckfro

### CODE REVIEW

File: app/controllers/auth/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Auth::SessionsController < ApplicationController
+
+  def new
+    @user = User.new
+  end
```
Comment (pinifloyd) [inline app/controllers/auth/sessions_controller.rb:4]: Hm... It seems that it doesn't meter. Well, I'll check it a few later.

File: config/initializers/sorcery.rb [hunk 2]
```diff
@@ -0,0 +1,463 @@
+# The first thing you need to configure is which modules you need in your app.
+# The default is nothing which will include only core features (password encryption, login/logout).
+# Available submodules are: :user_activation, :http_basic_auth, :remember_me,
+# :reset_password, :session_timeout, :brute_force_protection, :activity_logging, :external
+Rails.application.config.sorcery.submodules = []
```
Comment (pinifloyd) [inline config/initializers/sorcery.rb:5]: Will include submodules later, as at the moment I can't understand what exactly we need.

File: app/controllers/auth/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  def create
+    if @user = login(params[:email], params[:password])
+      redirect_back_or_to '/'
+    else
+      p @user.errors
```
Comment (ErrorSoul) [inline app/controllers/auth/sessions_controller.rb]: for debug?

Comment (pinifloyd) [inline app/controllers/auth/sessions_controller.rb]: Damn, right :+1: I'll remove it.

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -5,10 +5,19 @@
...
-    <%= stylesheet_link_tag    'application', media: 'all' %>
+    <%= stylesheet_link_tag 'application', media: 'all' %>
   </head>
 
   <body>
+    <!-- TODO: Temporary auth links here, will be removed -->
+    <% if current_user %>
+      <%= link_to 'logout', destroy_user_session_path %>
+    <% else %>
+      <%= link_to 'Login', new_user_session_path %>
```
Comment (ErrorSoul) [inline app/views/layouts/application.html.erb:16]: what we are doing with translate or not now (other task)? 

I think we can add I18n.t for now, but without .yml files

Comment (pinifloyd) [inline app/views/layouts/application.html.erb:16]: It's just WIP :) So yes, we should use translates. Why without .yml? We can add it right now )

**Comment by ErrorSoul** [comment]
Look's good. Sorcery is more understable, than devise :)

**Comment by pinifloyd** [comment]
Yes, think so :)

**Pull request merged**